pymaker/__init__.py

```diff
@@ -148,9 +148,10 @@ def _get_contract(web3: Web3, abi: list, address: Address):
 
         return web3.eth.contract(abi=abi)(address=address.address)
 
-    def _past_events(self, contract, event, cls, number_of_past_blocks, event_filter) -> list:
+    def _past_events(self, contract, event, cls, number_of_past_blocks, event_filter, to_block = None) -> list:
```

Comment:
I guess we can think about improving it by introducing a single but more object-oriented parameter. Like an abstract `EventBlockRange` interface with `PastBlocks(no_of_blocks)` and `BlockRange(from_block, to_block)` implementations. They could both implement `from_block(web3: Web3)` and `to_block(web3: Web3)`.

Otherwise it can get really messy with all these parameters around the place, especially if we come up with some more types of behaviour in the future.